### DISCUSSION TITLE
one cluster two flux2 installations in different namespaces

### DISCUSSION THREAD

**Discussion by tibberg** [fluxcd/flux2#1148] [Q&A] [answered]
We have a single test cluster where two independent teams bootstrapped flux (both teams use `--watch-all-namespaces=false`). We don't want to store our resources in the same git repo. While they were just experimenting with flux but we're using it since a while now. They tested how the `flux uninstall` worked, and even if they bootstrapped flux again, this rendered our flux installation  not working, as all of the custom resources were deleted. We had to re-bootstrap flux again to get it working. 
This is not a surprise, as the CRDs are cluster-wide, not tied to the namespace. 
But the question is: 
what are the best practices to get two teams work with the same cluster?

**Comment by stefanprodan** [comment, marked as answer]
Flux should be installed only once per cluster. Please see here how to manage multi teams/repos/apps https://github.com/fluxcd/flux2-multi-tenancy

**Reply by tibberg** [reply]
Ok, I see. Now my question is: how dangerous is the current practice we're running right now? How urgent is the switch to the multitenant model? I've read the docs quoted, but we're using helm exclusively, and I don't know about kustomize, so it is hard for me to understand what steps do we need to take exactly to convert the current setup to multitenant.

**Reply by stefanprodan** [reply]
I see no urgency as long as you don't use `flux uninstall`. To remove one instance, delete the flux deployments and services and it should have no affect on any other instance.
